### PR TITLE
recreate questions

### PR DESCRIPTION
**Pull request by oleg08**
rebase #6

rebase with master

### CODE REVIEW

File: app/controllers/questions_controller.rb [hunk 1]
```diff
@@ -4,23 +4,56 @@ def index
   end
 
   def show
-    if current_user.admin?
-      if params[:id].to_i > 0 && params[:id].to_i <= Question.all.count  
+    #if current_user.admin?
```
Comment (serhiym) [inline app/controllers/questions_controller.rb]: remove commented code if we don't need it

File: db/migrate/20151201142313_add_index_to_questions_content.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToQuestionsContent < ActiveRecord::Migration
+  def change
+    add_index :questions, :content, unique: true
```
Comment (serhiym) [inline db/migrate/20151201142313_add_index_to_questions_content.rb]: why do we need index questions content?

File: app/models/question.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 class Question < ActiveRecord::Base
 	belongs_to :topic
```
Comment (serhiym) [inline app/models/question.rb]: use 2 spaces indentation
+ good approach to not use validates for foreign fields *_id
try to find another way how you can write this code

File: app/views/questions/_form.html.erb [hunk 4]
```diff
@@ -1,11 +1,14 @@
 <%= form_for(@question) do |f| %>
-<%= f.label :content, "Question" %>
-<%= f.text_field :content %>
-<%= f.label :answer, "Answer" %>
-<%= f.text_field :answer %>
+    <%= render 'shared/error_messages', object: f.object%>
```
Comment (serhiym) [inline app/views/questions/_form.html.erb]: 2 spaces

File: app/views/shared/_error_messages.html.erb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<% provide(:object, params[:object]) %>
```
Comment (serhiym) [inline app/views/shared/_error_messages.html.erb]: you don't need this file and render errors in form partial
bug with flash error show was fixed - please rebase with the latest code from master

File: db/migrate/20151201142313_add_index_to_questions_content.rb (hunk 2 above)
Comment (oleg08) [inline db/migrate/20151201142313_add_index_to_questions_content.rb]: we need index questions content to enforce uniqueness in database level

Comment (serhiym) [inline db/migrate/20151201142313_add_index_to_questions_content.rb]: we don't need it - it's bad approach
please remove this migration

File: db/migrate/20151201142325_add_index_to_topics_title.rb [hunk 6]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToTopicsTitle < ActiveRecord::Migration
+  def change
+    add_index :topics, :title, unique: true
```
Comment (serhiym) [inline db/migrate/20151201142325_add_index_to_topics_title.rb]: remove this migration

**Pull request merged**